app_internal_test.go

```diff
@@ -115,3 +117,37 @@ func TestAnnotationError(t *testing.T) {
 	assert.ErrorIs(t, err, wantErr)
 	assert.Contains(t, err.Error(), wantErr.Error())
 }
+
+// TestDisableSignalHandler tests disabling the default signal handler.
+// It's an internal test to allow us replace the app.receivers.notify function.
+func TestDisableSignalHandler(t *testing.T) {
+	for _, disableSignalHandler := range []bool{false, true} {
+		testName := "Not disabling signal handler"
+		if disableSignalHandler {
+			testName = "Disabling signal handler"
+		}
+		t.Run(testName, func(t *testing.T) {
+			app := New()
+			calledNotify := false
+
+			app.receivers.notify = func(c chan<- os.Signal, sig ...os.Signal) {
+				calledNotify = true
+			}
+
+			if disableSignalHandler {
+				opt := DisableSignalHandler()
+				opt.apply(app.root)
+			}
+
+			app.Start(context.Background())
+			app.Stop(context.Background())
+
+			if disableSignalHandler {
```

Comment:
assert.Equal(t, disableSignalHandler, !calledNotify)

No if/else needed, no ?

Reply:
yes, I thought this was clearer. I have no real preference here. Happy to change if maintainers would prefer.